Decentralize the dev fee (#278)
zcash/zips: zip-XXXX.rst

leading to a more resilient network, stronger technology, and wider usage.

The Zcash narrative
-------------------

We're all here to build a robust, private, decentralized currency. But in the
dev fee proposals I've seen so far, the idea of a Zcash narrative that
distinguishes it from the competition is absent.

Of the slew of ZIPs addressing Zcash's future, there's only been one strong
narrative case -- the idea that Zcash exists purely as a hedge against Bitcoin's
long-term privacy failure. Put simply, Zcash is "Bitcoin, but private".

Zcash should aim higher. Bitcoin is the only coin that has successfully made a
store of value argument, which I like to call "worse is better". Don't upgrade
the network -- the argument goes -- stability is more important than solving
today's problems.  Bitcoin is chasing the `Lindy
effect<https://en.wikipedia.org/wiki/Lindy_effect>`, where worse is better, and
the network becomes stronger every day it survives. That's great for Bitcoin.
For the rest of us, though, better is better. Zcash *should be better*.

Zcash is known for having the best tech in the space, built by one of the best

> team's in the space. We should lean in to that reputation, nurturing the best

Reviewer: ```suggestion
teams in the space. We should lean in to that reputation, nurturing the best
```

research and engineering talent to take Zcash to the next level, and leveraging
a Zcash dev fee as a differentiator to build the world's best private medium of
exchange.

Principles of cryptocurrency governance
---------------------------------------

To understand Zcash governance, it's worth reviewing "default" cryptocurrency
governance. Most proof-of-work chains today have three major governing roles:

1. Miners validate and secure the chain. They do some work to earn a reward.
   Miners are the first owners of newly minted coins, and are an integral part
   of network upgrades.
2. Users buy, hold, and spend the currency. In networks like Bitcoin, they also
   run full nodes, strengthening network resilience by decentralizing
   validation.
3. Developers maintain clients to power and interact with the network. They
   typically propose network upgrades.

On a chain like Bitcoin, any of these roles can veto a network upgrade.

1. Miners can veto activating a new fork by refusing to build off blocks using
   new network rules, orphaning a minority effort. They can also attack any fork

>    attempt that doesn't change

Reviewer: ```suggestion
   attempt that doesn't change.
```

2. Users can veto a fork by refusing to update their full nodes, rejecting
   blocks as invalid -- as demonstrated in the UASF fiasco resulting from the

>    SegWit2x attempt to force a Bitcoin hardfork. Users can also vote with their

Reviewer: ```suggestion
   Bitcoin miners' refusal to activate SegWit. Users can also vote with their
```

Reviewer: Correction: UASF (via BIP-148) was first proposed before Segwit2x was announced, and was in response to the miners' ongoing refusal to activate SegWit.

Author: Ofc, good catch!

   might not seem like a strong veto, in practice that means any fork will need
   at least one additional development team, or the agreement of existing client
   software developers.

These roles together form a balance of power that makes contentious forks
difficult -- any change that a large swath of users disapproves of could split
the chain.

The state of play
-----------------

In Zcash, the addition of the Electric Coin Company (ECC) and the Zcash
Foundation skew this balance.

Both organizations are comprised of Zcash founders, developers, researchers, and
privacy proponents who have driven this ecosystem forward and represent its
values. Nevertheless, their mode of operation today skews a healthy balance of
power in Zcash governance.

The mechanisms of that skew are the Zcash trademark, held by the ECC, and
primary client software development, now split between the ECC and the
Foundation.

In a disagreement between miners, users, and developers, the ECC has the
unilateral option of enforcing the Zcash trademark, effectively allowing them
to choose a winning fork against the will of users, miners, and other

> developers.

Reviewer: This section should probably be updated to reflect the [new trademark deal](https://www.zfnd.org/blog/zcash-trademark-resolution/).

While the Foundation's maintenance of the `zebrad` client would normally allow
them to "soft veto" a network upgrade, they don't have a similar veto on the

> Zcash trademark enforcement.

Reviewer: This section should probably be updated to reflect the [new trademark deal](https://www.zfnd.org/blog/zcash-trademark-resolution/).

Compounding these issues, the Foundation and the ECC aren't arms-length entities
as they're organized today.

This situation poses a number of problems for new and existing Zcash users, as
well as both entities.

* The threat of a central entity overriding (or being forced to override) the
  will of users undermines self-sovereignty.
* The ECC and Foundation are both put at legal risk. As entangled entities,
  they're remarkably similar to a single entity when trying to minimize
  regulatory risk.
* Power between the two entites *hasn't* been decentralized. The ECC remains

>   a unilateral power, as well as a single point of failure.

Reviewer: Somewhat different now without monopoly trademark rights. (ref: https://www.zfnd.org/blog/zcash-trademark-resolution/).

and governance, specifying the structure beyond the next 4 years is out of
scope.  Much will have changed in 4 years. Perhaps this structure will be
sufficient; perhaps we'll be battling the Third Crypto War, and need to go back
to the drawing table.

Specification
=============

The below proposal is an effort to cleanly resolve the problems with Zcash's
current governance, while

* maintaining a balance of power between stakeholders
* removing single points of failure / control
* growing development and usage of Zcash
* and supporting the best interests of miners, users, and developers *today*.

Decentralizing development
--------------------------

A few proposals have suggested the introduction of a mysterious "third entity"
to resolve disagreements between the Foundation and the ECC.

I prefer a different approach, refocusing the role of the Foundation and making
room for the ECC to work with a robust developer ecosystem.

In this proposal, the Foundation shall support community development through
running the forum and events, gathering community sentiment, managing short-term
development grants, and conducting the diligence behind the assignment and

> disbursement of a development fee. This development fee shall be funded by 20%

Author: That is the intent, yes.

Separating the people disbursing funds from the people doing work is a core goal of this proposal. Getting into the weeds, I think `zcashd` should be deprecated and eventually be sunset, post NU4 the Foundation should hand off `zebrad` to a professional maintainer, and another dev fee recipient should start a new client.

Our team is interested in building and maintaining a Go client.

of the block reward, with as much as half of the fee burned in case of
extraordinary growth in the price of ZEC.

The Foundation shall receive 25% of the dev fee. If the volume-weighted average
price of ZEC over the month means the foundation would receive greater than

> $500k that month, the Foundation shall burn enough ZEC such that their max

Author: @daira Here's the example of a burn mechanism where I'd rather actually "give all excess ZEC back to the miners pro-rata in a 30-day period", but I was worried about implementation complexity, especially in a post t-address world. Thoughts on a simple proposal that's forward-compatible?

benefit is

.. math::

  MaxBenefit(RewardDollarAmount) = Min(500000, 500000 * \sqrt{\frac{RewardDollarAmount/500000}})

Capping the monthly upside of the Foundation will limit growth, while

> encouraging fiscal discipline.

Reviewer: If the funds are spent on development (and marketing, and community management, etc) then are these funds really only to the Foundation's benefit, as implied? Aren't they really to be spent for the benefit of the whole network?

In a company, usually it's expected that operational costs increase as revenues increase to fuel growth. Why shouldn't the Zcash network similarly be able to scale up spending on the Dev Fund (and all of its funded activities) as the ZEC price increases, without this artificial ceiling capping spending on the Fund?

I see your stated reasoning:

> Capping the monthly upside of the Foundation will limit growth, while encouraging fiscal discipline.

But I don't agree that the former is desirable, or that it's worth doing in pursuit of the latter. I want to dig into the reasoning more since burning money that could otherwise be spent directly on growth activities while the network is clearly going through a growth phase (ZEC price rising substantially) seems like a very consequential decision.

Author: > In a company, usually it's expected that operational costs increase as revenues increase to fuel growth. Why shouldn't the Zcash network similarly be able to scale up spending on the Dev Fund (and all of its funded activities) as the ZEC price increases, without this artificial ceiling capping spending on the Fund?

We just saw what that did to the ECC, instead -- it led to unsustainable spending and worse fiscal discipline, followed by a rude awakening. This isn't revenue the ZF has "grown", regardless, so I'm not sure the analogy holds.

The idea here is that we want to keep the operation overhead of the ZF capped to promote low overhead / bureaucracy and capital efficiency. I am about to make a couple tweaks though

> I want to dig into the reasoning more since burning money that could otherwise be spent directly on growth activities while the network

About to remove the burn in favor of a longer-term purpose-restricted fund.

The remaining 75% of the dev fee shall be distributed between development teams
working to maintain clients.

* One third of the remaining fee (25% of the total) shall be reserved for the
  role of the "principal developer", a developer with additional voice in Zcash
  governance. The principal developer allocation shall be capped similarly to

>   the Foundation's, based on the monthly volume-weighted average price.

Author: They don't meet. $700k / month is too stiff a burn, and they should cut back unless market conditions seriously change. Note that even if things go well, this proposal still suggests they cut back with a burn function that kicks in before $700k / month.

There is an opportunity here for the ECC to split into distinct entities, though.

* The remaining two thirds of the fee (50% of the total), called the "outside
  development fee", shall be distributed between at least two development teams,
  chosen semi-annually by the Foundation, coinciding with network upgrades.
  Unlike those of the Foundation and principal developer, these allocations

>   aren't affected by market conditions, and don't carry a burn requirement.

Author: We're one of them- I'll be following up shortly with a Go client proposal, as well as an interop project proposal from an adjacent org

well as monthly progress updates.

All work funded by the dev fee will be open source, under licenses compatible
with the existing Zcash clients.

Though the Foundation shall periodically judge the efficacy of dev fee
recipients, deciding on and driving roadmaps for Zcash is the role of dev fee
recipients, in partnership with the community. This role is neatly balanced by
users running full nodes and miners, either of which can veto a network upgrade.

While dev fee recipients are not required to work exclusively on Zcash,
considering the nature of their work, recipients must guarantee they aren't
obliged to the interests of competing projects.

The role of the principal developer
-----------------------------------

The role of the principal developer is as a "first among equals" amongst the dev
fee recipients.

The principal developer shall make a number of guarantees.

1. Zcash shall be their exclusive focus, submitting financials periodically to
   the Foundation as assurance.
2. They shall maintain a well-run board and employ a qualified CFO.
3. In addition to the existing open-source requirements, they shall agree to

>    assign any trademarks or patents relevant to Zcash to the Foundation.

Author: Yep. I think that's a poor idea because trademarks are use-it-or-lose-it. We need a single entity deciding to enforce, and deciding when to give up the :tm:

In exchange, the principal developer is granted an indefinite dev fee allocation
and a wide remit to pursue longer-term research relevant to Zcash, as well as

> a voice on the board of the Foundation. The principal developer will not be

Reviewer 2: Are you suggesting that the Principal Developer representative *will serve on the board of a non-profit Foundation* which *can revoke its funding*? This looks like a clear CoI and probably disallowed.

Author: I am suggesting that, yes- the implication being that they step out whenever the CoI comes up, and we have a 4-of-4 decision regarding the principal developer.

> This looks like a clear CoI and probably disallowed.

I'm... skeptical given the current situation that this scenario is worse, or breaking any rules around 501(c)(3) orgs. Happy to get legal advice if it helps move the conversation along though.

subject to semi-annual review, though it will submit the same plans required of
other recipients. The principal developer will only be recommended for removal
by the Foundation in extraordinary circumstances, including reneging on the
above guarantees or extreme negligence.

Minimum viable Foundation
-------------------------

To manage the dev fee and fulfill its community and diligence duties, the
Foundation shall maintain a board of 5 independent members. Rather than the
structure in the current bylaws, the board will consist of

> * 1 seat voted on by ZEC holders directly.

Reviewer 2: How? We don't have a such a system in place, and creating one is nontrivial R&D.

Author: 100%, I've kicked the can here. I also know we'd be open to helping with this

Author: In general, I'm looking to the Foundation to provide a plan to execute most of this- that's why I haven't included timelines and left a number of other governance questions unanswered. I think the Foundation can step up.

> * 1 seat representing a newly created research advisory board, whose primary

Reviewer 2: Who serves on the "research advisory board"? How are they chosen and replaced?

Author: I think they're self-selected initially. I'm looking to change the by-laws as little as possible, while also making a clear place for founding scientists and other researchers who have an interest in Zcash's success and can help with diligence.

  role will be technical diligence of potential recipients of the dev fee.
* 1 seat for the "principal developer", a privileged recipient of the Zcash
  dev fee acting as "first among equals" amongst a variety of dev fee recipients
  building on Zcash.
* 2 seats elected by the board, as the board is currently selected according to
  the bylaws. The board's discretion here means these could be selected via a
  community election, or via the remaining 3 seats' direct vote.

The Foundation requires a professional board. Board member selection should
heavily favor candidates with existing formal public or private sector board
experience. Each board member shall be paid reasonably by the Foundation.

Each board member should bring a unique network and set of skills to bear to
increase the impact of the Foundation.

> Outside the seat for the principal developer, no board members shall have an

Reviewer 2: Since the whole point of this system is to decentralized the Dev Fee and have it support numerous players in the ecosystem, directly and the grants, we may reach a point where most of the people who are technically cognizant and aligned with Zcash's mission will be tainted by Dev Fund. (Arguably, this is already the case!) So your requirement here effectively excludes BUIDLers from the board, leaving it just to financial interests and outsiders.

Author: I don't know, between academia and holders I think there should be plenty of people with interest and skin in the game?

I'm also loosely assuming fewer than 10 recipients of the fee for the medium-term. Whether that's wise is worth debate, but I'm looking to engender a situation similar to Ethereum's multi client implementation teams
